R/tm_a_pca.R

```diff
@@ -294,11 +294,10 @@ srv_a_pca <- function(id, data, reporter, filter_panel_api, dat, plot_height, pl
       response[[i]]$select$multiple <- FALSE
       response[[i]]$select$always_selected <- NULL
       response[[i]]$select$selected <- NULL
-      response[[i]]$select$choices <- var_labels(isolate(data())[[response[[i]]$dataname]])
-      response[[i]]$select$choices <- setdiff(
-        response[[i]]$select$choices,
-        unlist(teal.data::join_keys(isolate(data()))[[response[[i]]$dataname]])
-      )
+      all_cols <- var_labels(isolate(data())[[response[[i]]$dataname]])
+      ignore_cols <- unlist(teal.data::join_keys(isolate(data()))[[response[[i]]$dataname]])
+      color_cols <- all_cols[!names(all_cols) %in% ignore_cols]
+      response[[i]]$select$choices <- choices_labeled(names(color_cols), color_cols)
```

Comment:
```suggestion
      all_cols <- colnames(isolate(data())[[response[[i]]$dataname]])
      ignore_cols <- unlist(teal.data::join_keys(isolate(data()))[[response[[i]]$dataname]])
      color_cols <- all_cols[!all_cols %in% ignore_cols]
      response[[i]]$select$choices <- choices_labeled(color_cols, color_cols)
```

Reply:
@vedhav sorry this is not  good suggestion. But I have a feeling that it can be still simplified

Reply:
I just made use of the var labels. Please pull the branch, as these labels are retained.